os/fs/smartfs/smartfs_utils.c

```diff
@@ -1036,6 +1036,65 @@ int smartfs_unmount(struct smartfs_mountpt_s *fs)
 	return ret;
 }
 
+/****************************************************************************
+ * Name: smartfs_get_datalen
+ *
+ * Description: Calculates the length of the opened file
+ *
+ ****************************************************************************/
+
+int smartfs_get_datalen(struct smartfs_mountpt_s *fs, uint16_t firstsector, uint32_t *datalen)
+{
+	int ret;
+	uint16_t dirsector;
+	struct smart_read_write_s readwrite;
+	struct smartfs_chain_header_s *header;
+
+	(*datalen) = 0;
+
+	header = (struct smartfs_chain_header_s *)fs->fs_rwbuffer;
```

Comment:
Is it okay not to check that `fs` is NULL or not?
If NULL, it can be dereferencing.

Reply:
We do not need to check for the mountpoint structure as it is passed down from many parent functions, and is checked for validity at the top of that tree.